src/files/BrsFile.ts

```diff
@@ -121,14 +138,91 @@ export class BrsFile {
         }
     }
 
-
-    public parser: Parser;
+    public parser = new Parser();
 
     public fileContents: string;
 
+    private logDebug(...args) {
+        this.program.logger.debug('BrsFile', chalk.green(this.pkgPath), ...args);
+    }
+
+    /**
+     * If this is a typedef file
+     */
+    public isTypedef: boolean;
+
+    /**
+     * The key to find the typedef file in the program's files map.
+     * A falsey value means this file is ineligable for a typedef
+     */
+    public typedefKey?: string;
+
+    /**
+     * If the file was given type definitions during parse
+     */
+    public hasTypedef;
+
+    /**
+     * A reference to the typedef file (if one exists)
+     */
+    public typedefFile?: BrsFile;
+
+    /**
+     * An unsubscribe function for the dependencyGraph subscription
+     */
+    private unsubscribeFromDependencyGraph: () => void;
+
+    /**
+     * Find and set the typedef variables (if a matching typedef file exists)
+     */
+    private resolveTypdef() {
+        this.typedefFile = this.program.getFileByPathAbsolute<BrsFile>(this.typedefKey);
+        this.hasTypedef = !!this.typedefFile;
+    }
+
+    /**
+     * Attach the file to the dependency graph so it can monitor changes.
+     * Also notify the dependency graph of our current dependencies so other dependents can be notified.
+     */
+    public attachDependencyGraph(dependencyGraph: DependencyGraph) {
+        if (this.unsubscribeFromDependencyGraph) {
+            this.unsubscribeFromDependencyGraph();
+        }
+
+        //anytime a dependency changes, clean up some cached values
+        this.unsubscribeFromDependencyGraph = this.program.dependencyGraph.onchange(this.dependencyGraphKey, () => {
+            this.logDebug('clear cache because dependency graph changed');
```

Comment:
Log message is unclear now that here isn't a cache

Reply:
resolved.